database/src/main/java/com/firebase/ui/database/OnIndexMismatchListener.java

```diff
@@ -0,0 +1,7 @@
+package com.firebase.ui.database;
+
+import com.google.firebase.database.DataSnapshot;
+
+interface OnIndexMismatchListener {
```

Comment:
Yup. That first snippet looks like it. Not sure about the second snippet though. What would the `mJoinResolver` consist of there?